sdk/eventhub/azure-eventhub/samples/async_samples/recv_with_checkpoint_by_time_interval_async.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python
+
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+"""
+An example to show receiving events from an Event Hub with checkpoint store doing checkpoint
+by every fixed time interval asynchronously.
+In the `receive` method of `EventHubConsumerClient`:
+If no partition id is specified, the checkpoint_store are used for load-balance and checkpoint.
+If partition id is specified, the checkpoint_store can only be used for checkpoint.
+"""
+
+import asyncio
+import os
+import time
+from azure.eventhub.aio import EventHubConsumerClient
+from azure.eventhub.extensions.checkpointstoreblobaio import BlobCheckpointStore
+
+CONNECTION_STR = os.environ["EVENT_HUB_CONN_STR"]
+STORAGE_CONNECTION_STR = os.environ["AZURE_STORAGE_CONN_STR"]
+BLOB_NAME = "your-blob-name"  # Please make sure the blob resource exists.
+
+
+partition_last_checkpoint_time = dict()
+checkpoint_time_interval = 15
+
+
+async def on_event(partition_context, event):
+    # put your code here
+    p_id = partition_context.partition_id
+    print("Received event from partition: {}".format(p_id))
+    now_time = time.time()
+    p_id = partition_context.partition_id
+    if (partition_last_checkpoint_time.get(p_id) is None) or \
+            (now_time - partition_last_checkpoint_time.get(p_id)) >= checkpoint_time_interval:
```

Comment:
Minor: For optimum performance, we only want to do the dict.get once:
```
current_checkpoint = partition_last_checkpoint_time.get(p_id)
if current_checkpoint is None or (now_time - current_checkpoint) >= checkpoint_time_interval:
    ...
```

Reply:
updated :)